src/main/java/openfasttrack/importer/specobject/ImportHelper.java

```diff
@@ -55,36 +67,73 @@ void runImport() throws XMLStreamException
 
             if (currentEvent.isStartElement())
             {
+                if (skipFile(currentEvent.asStartElement()))
+                {
+                    return;
+                }
                 foundStartElement(currentEvent.asStartElement());
             }
         }
     }
 
+    private boolean skipFile(final StartElement startElement)
+    {
+        if (this.validRootElementFound)
+        {
+            return false;
+        }
+        if (rootElementWhitelisted(startElement))
+        {
+            this.validRootElementFound = true;
+            return false;
+        }
+        LOG.warning(() -> "Found unknown XML root element '" + startElement.getName()
+                + "': skip file.");
+        return true;
+    }
+
+    private boolean rootElementWhitelisted(final StartElement rootElement)
+    {
+        final String elementName = rootElement.getName().getLocalPart();
+        return SPECDOCUMENT_ROOT_ELEMENT_NAME.equals(elementName);
+    }
+
     private void foundStartElement(final StartElement element) throws XMLStreamException
     {
         switch (element.getName().getLocalPart())
```

Comment:
We need the StartElement later in `foundStartElement()` to get attributes from it. I would prefer not to add an parameter for the element name with duplicate information.